### ISSUE TITLE
read -t html causes plain text emails to be blank

### ISSUE THREAD

**Issue by callumacrae** [pimalaya/himalaya#352]
Disclaimer: I don't know if this is a himalaya thing or an IMAP thing - but either way, it would be nice for himalaya to be able to get around it

I _generally_ want my emails to be read as HTML as I'm piping them into `w3m`, but this doesn't work for plain text emails - it just displays an empty message instead

```
❯ himalaya read 232
hello this is a test email


❯ himalaya read 232 -t html


```

Ideally there would be some way of having himalaya return HTML where possible, otherwise fall back to plain text

**Comment by soywod** [comment]
`himalaya read 232` means "Please, I want to read the message 232. If
you do not find the plain part then show me a readable version of the
HTML part". It does so by "purifying" the HTML.

`himalaya read 232 -t html` means "Please, I really want the HTML part
of the message 232 so I can pipe the result into w3m". Maybe it is a bit
too strict. I think Himalaya should fallback to the plain part if it
does not find the HTML part. What happens if you pipe a plain part to
`w3m`?

On 29 Mar 2022, at 18:50 (+0200), soywod/himalaya wrote:
> Disclaimer: I don't know if this is a himalaya thing or an IMAP thing - but either way, it would be nice for himalaya to be able to get around it
> 
> I _generally_ want my emails to be read as HTML as I'm piping them into `w3m`, but this doesn't work for plain text emails - it just displays an empty message instead
> 
> ```
> ❯ himalaya read 232
> hello this is a test email
> 
> ❯ himalaya read 232 -t html
> 
> ```
> 
> Ideally there would be some way of having himalaya return HTML where possible, otherwise fall back to plain text
> 
> -- 
> Reply to this email directly or view it on GitHub:
> https://github.com/soywod/himalaya/issues/352
> You are receiving this because you are subscribed to this thread.
> 
> Message ID: ***@***.***>

-- 
Cordialement
Clément DOUIN
Développeur Web Full-Stack
https://soywod.me

**Comment by callumacrae** [comment]
> What happens if you pipe a plain part to `w3m`?

The links aren't links and there are no images!

**Comment by soywod** [comment]
> The links aren't links and there are no images!

So I guess we can safely fallback to the plain part if no HTML part is
found. Thanks for you feedback!

On 29 Mar 2022, at 22:07 (+0200), soywod/himalaya wrote:
>> What happens if you pipe a plain part to `w3m`?
> 
> The links aren't links and there are no images!
> 
> -- 
> Reply to this email directly or view it on GitHub:
> https://github.com/soywod/himalaya/issues/352#issuecomment-1082325519
> You are receiving this because you commented.
> 
> Message ID: ***@***.***>

-- 
Cordialement
Clément DOUIN
Développeur Web Full-Stack
https://soywod.me

**Comment by pleshevskiy** [comment]
I like plain version, but i prefer to get all links. At the moment I have to open html mime-type to find links...

**Comment by soywod** [comment]
Does this PR https://github.com/soywod/himalaya/pull/357 fixes your issue? If so then you can close it, it should be fixed for the next release :wink:

**Comment by soywod** [comment]
It looks like the issue #357 does not fix this one. I will try to fix that as soon as I can then!

**Comment by soywod** [comment]
Fixed in the last release [`v0.6.1`](https://github.com/soywod/himalaya/releases/tag/v0.6.1) :tada:
